### DISCUSSION TITLE
Run without BladePitch Controller

### DISCUSSION THREAD

**Discussion by oguzcrs11** [OpenFAST/openfast#2449] [Q&A] [answered]
Dear @jjonkman ;

I need your help again for a different issue. For 5mw I use the fst and dll files in rtest
I want to run it at constant rpm with shaft brake without pitch control.
Pitch control inactive and the shaft brake active
![{4A1B6DBC-74D9-4E3E-928B-91AEB108F557}](https://github.com/user-attachments/assets/64e0a1ff-4ce1-411c-a33d-1b84b33c8212)

buti still rpm increases with time
![{B4E2FF3F-ECF5-49FE-9EF2-F634F5B79DE5}](https://github.com/user-attachments/assets/a1f5a0b0-818e-4891-98ed-a20243b33b53)

Simulation was run for the NREL 5MW and operating condition of 12.1 rpm at 12 m/s speed.

![{EA752FBB-AA37-4603-82C2-63A715C184CA}](https://github.com/user-attachments/assets/3e0e7c32-59fc-4c3c-a9bb-8aa19a8d988d)

![{A6AE47E9-373E-4C91-BAFA-8858881BD851}](https://github.com/user-attachments/assets/54d1929d-5865-488e-af60-849187a84b14)

In the first image, when pitch control is active, constant tsr with time , in the second image, pitch is not active and brake control is active.
What is your advice for me to get fixed rpm working conditions without pitch being active

Best Regards.

**Comment by jjonkman** [comment]
Dear @oguzcrs11,

I'm not understanding everything you are saying, but I see that you selected `HSSBrMode` = 5, which means that the HSS brake deployment is controlled by the Bladed-style DLL, but the DLL you are likely using for the NREL 5-MW baseline turbine (if you are using the default DLL) is not set up to regulate HSS brake deployment.  You can deploy the HSS brake without the DLL by setting `HSSBrMode` = 1 followed by setting of the simple HSS brake parameters.

Best regards,

**Comment by oguzcrs11** [comment]
Dear @jjonkman,

FAST is really a vast and comprehensively considered tool - I couldn't fully explain my problem, I apologize.

If I am interpreting correctly, the pitch angle control in ServoDyn functions as a braking system, but there should also be a mechanical braking system in system I guess also it's "torque and generator control". In the model I have created, I aim to operate the system at a constant TSR without activating the pitch controller I'm actually trying to figure out if I can do this with generator and torque control, so without different pitch angles values that I can compare my results with a simple CFD model. Do you think planning such a setup within FAST for the simulation is reasonable "trying to run the system with the pitch control deactivated" ?

Best Regards.

**Reply by jjonkman** [reply]
Dear @oguzcrs11,

You can certainly run OpenFAST without pitch control enabled.  Pitch control is normally active only in control region 3 (between rated and cut-out wind speeds) anyway.  Maintaining constant TSR in control region 2 (between cut in and rated wind speeds) is typically handled through torque control.

Best regards,

**Reply by oguzcrs11** [reply]
Dear @jjonkman,

Thanks a lot for your reply. Considering the ServoDyn Inputs, I think that the values ​​in the image below will change at different wind speeds and rpm. 

![image](https://github.com/user-attachments/assets/d07a0327-8abc-4b03-9823-eb41c915f3ec)

Is there a study or documentation where I can get information about the calculation of these variables?

Best Regards.

**Reply by jjonkman** [reply, marked as answer]
Dear @oguzcrs11,

Presumably you are referring to the simple torque = k*omega^2 control law often used in control region 2 that is available through the simple variable-speed torque controller (`VSContrl` = 1 option of ServoDyn).  For a discussion of the ServoDyn variables used by this controller, see the following topic on our forum: https://forums.nrel.gov/t/simulation-wind-turbine-behaviour/1960 (especially my post dated Jan 14, 2019).

Best regards,

**Reply by oguzcrs11** [reply]
Dear @jjonkman,

Thank you for your reply.

Best Regards.

**Reply by oguzcrs11** [reply]
Dear @jjonkman,

As your advice, I applied torque = _k*omega^2_  for _VSContrl = 1_ option of ServoDyn ınputs;

Gentorque= 97*12.1 (12.1 initial rotor speed * gearbox ratio) ----> rpm to rad/s = 122.88639
VS_RtTq = 5000000/122.88639 = 40687.988317 (N-m)
VS_Rgn2K = 40687.988317 / (122.88639)^2 = 2.6943787556
![image](https://github.com/user-attachments/assets/bb5157d4-7f60-4748-abf3-31c46100f35e)

While I wanted the system to remain stable at 12.1 rpm, I encountered a value like 13.56

column name : Time - RtSpeed - TSR
![image](https://github.com/user-attachments/assets/88b08279-d644-4dc3-9f88-652248bde73c)

Could I be doing something wrong, do you have any advice for this?
Thank you in advance, Best Regards.

**Reply by jjonkman** [reply]
Dear @oguzcrs11,

I'm not sure I understand what are trying to do.  Is there a reason why you are replacing the baseline torque and pitch controller provided for the NREL 5-MW baseline wind turbine (as implemented in a Bladed-style DISCON dynamic library)?

In your case, I gather you've replaced the baseline controller with a simple variable-speed control law, but disabled pitch control.  Given that this simple torque controller relies on constant torque above rated, pitch control would be required to maintain rotor speed above rated.  If you don't have pitch control enabled, the rotor will overspeed as you are demonstrating.

Best regards,

**Reply by oguzcrs11** [reply]
Dear @jjonkman,
First of all, thank you for your valuable answers.

Yes, I considered achieving a constant TSR by performing torque control without pitch control, and I wanted to compare my values with the existing ones. This way, I could produce solutions for different pitch angles in the simulation, which remained constant in each simulation.

Then I tried to get solution by using dll file as you suggested 0 pitch degree (without pitch control) and, under a timeseries containing speed data from 5 m/s to 25m/s. When I compared my results with the results in
https://github.com/NREL/ROSCO/blob/main/Examples/Test_Cases/NREL-5MW/Cp_Ct_Cq.NREL5MW.txt 
file, I got these results.

![{8A176F65-0C81-40C1-A7F0-379FE0825BCB}](https://github.com/user-attachments/assets/61f1bc3e-89ef-48f3-8461-8fc417c8ecf0)

After establishing a stable structure and getting my values ​​close to the reference values, I want to test using different airfoils in the nrel 5mw turbine design. But I think there are too many differences between my values ​​and ı think that I am far from a stable simulation structure. Do you have any suggestions on this subject? I am using the sample file in r_Test. The only difference in this section is the wind file.

![{EBF38921-90AC-473C-99C8-75DA7A9CB9EA}](https://github.com/user-attachments/assets/6ab7bfd6-df17-4099-86de-60f6e5ecbadb)
![{3B3ACBB0-8489-4147-A87A-C531C7AF6A0F}](https://github.com/user-attachments/assets/55fbecde-18b7-44e2-b802-610698575139)
![{78B4CCEC-7696-408D-BE48-CA88A4CB552A}](https://github.com/user-attachments/assets/aef4bfcf-3fc4-48a8-85b7-bb6caaa0ddb9)
![{19F2FD68-A538-4952-B27F-6971C12B2F89}](https://github.com/user-attachments/assets/33327fbd-f141-4955-8354-143374fb88f9)
![{256C3779-C3D9-4D06-BD10-EE3B28A163FD}](https://github.com/user-attachments/assets/cfae447c-362f-4af3-b470-6f075c18d84c)

Thank you very much in advance, Best Regards.

**Reply by jjonkman** [reply]
Dear @oguzcrs11,

Sorry, but I'm not sure I understand enough about what you are doing to comment.

Best regards,
